src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -148,6 +153,22 @@ private static Map<String, Provider<? extends RandomGenerator>> createFactoryMap
                 .filter(p -> !p.type().isInterface())
                 .collect(Collectors.toMap(p -> p.type().getSimpleName(), Function.identity()));
         }
+
+        @SuppressWarnings("removal")
+        private static List<Provider<? extends RandomGenerator>> filterBuiltIn(
+                Map<?, Provider<? extends RandomGenerator>> map) {
+            Predicate<Provider<?>> onlyBuiltIn;
+            if (System.getSecurityManager() != null) {
+                // Don't parse annotation for user classes
+                onlyBuiltIn = p -> VM.isSystemDomainLoader(p.type().getClassLoader());
+            } else {
+                onlyBuiltIn = p -> true;
```

Comment:
Should this variable be renamed to something else? Because right now it does the opposite of it's name. i.e. When `onlyBuiltIn` is set to `true`, like here, the code a few lines below which filters the stream accepts all `RandomGeneratorFactory` implementations that have been found.

Reply:
Good idea. Naming is hard - and I don't have a good name. Maybe `allowType`?

Reply:
I think what you have with the filter call is fine. I think just renaming the `onlyBuiltIn` variable should be enough. Perhaps rename it to `acceptedProviders`?